salt/states/module.py

```diff
@@ -149,8 +206,120 @@ def wait(name, **kwargs):
 watch = salt.utils.alias_function(wait, 'watch')
 
 
-def run(name, **kwargs):
+@with_deprecated(globals(), "Oxygen", policy=with_deprecated.OPT_IN)
+def run(**kwargs):
     '''
+    Run a single module function or a range of module functions in a batch.
+    Supersedes `module.run` function, which requires `m_` prefix to function-specific parameters.
+
+    :param returner:
+        Specify a common returner for the whole batch to send the return data
+
+    :param kwargs:
+        Pass any arguments needed to execute the function(s)
+
+    .. code-block:: yaml
+      some_id_of_state:
+        module.xrun:
+          - network.ip_addrs:
+            - interface: eth0
+          - cloud.create:
+            - names:
+              - test-isbm-1
+              - test-isbm-2
+            - ssh_username: sles
+            - image: sles12sp2
+            - securitygroup: default
+            - size: 'c3.large'
+            - location: ap-northeast-1
+            - delvol_on_destroy: True
+
+
+    :return:
+    '''
+
+    if 'name' in kwargs:
+        kwargs.pop('name')
+    ret = {
+        'name': kwargs.keys(),
+        'changes': {},
+        'comment': '',
+        'result': None,
+    }
+
+    functions = [func for func in kwargs.keys() if '.' in func]
+    for func in functions:
+        if func not in __salt__:
+            ret['comment'] = "Module function '{0}' is not available".format(func)
+            ret['result'] = False
+        elif __opts__['test']:
+            ret['comment'] = "Module function '{0}' is set to execute".format(func)
+            ret['result'] = True
```

Comment:
Could potentially overwrite the `ret['comment']` & `ret['result']` from a previous loop iteration where it fails, depending on the content of the SLS file and the order of the dict.